test/powershell/Modules/Microsoft.PowerShell.Security/ConstrainedLanguageRestriction.Tests.ps1

```diff
@@ -0,0 +1,890 @@
+##
+
+function Get-RandomFileName
+{
+    [System.IO.Path]::GetFileNameWithoutExtension([IO.Path]::GetRandomFileName())
+}
+
+function Set-SystemLockDown
+{
+    Invoke-LanguageModeTestingSupportCmdlet -SetLockdownMode
+    $ExecutionContext.SessionState.LanguageMode = "ConstrainedLanguage"
+}
+
+function Remove-SystemLockDown
+{
+    Invoke-LanguageModeTestingSupportCmdlet -RevertLockdownMode
+    Invoke-LanguageModeTestingSupportCmdlet -EnableFullLanguageMode
+}
+
+if ($IsWindows)
+{
+    $code = @'
+
+    #region Using directives
+
+    using System;
+    using System.Globalization;
+    using System.Reflection;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Security;
+    using System.Runtime.InteropServices;
+    using System.Threading;
+    using System.Management.Automation;
+
+    #endregion
+
+    /// <summary>Adds a new type to the Application Domain</summary>
+    [Cmdlet("Invoke", "LanguageModeTestingSupportCmdlet")]
+    public sealed class InvokeLanguageModeTestingSupportCmdlet : PSCmdlet
+    {
+        [Parameter()]
+        public SwitchParameter EnableFullLanguageMode
+        {
+            get { return enableFullLanguageMode; }
+            set { enableFullLanguageMode = value; }
+        }
+        private SwitchParameter enableFullLanguageMode;
+
+        [Parameter()]
+        public SwitchParameter SetLockdownMode
+        {
+            get { return setLockdownMode; }
+            set { setLockdownMode = value; }
+        }
+        private SwitchParameter setLockdownMode;
+
+        [Parameter()]
+        public SwitchParameter RevertLockdownMode
+        {
+            get { return revertLockdownMode; }
+            set { revertLockdownMode = value; }
+        }
+        private SwitchParameter revertLockdownMode;
+        
+        protected override void BeginProcessing()
+        {
+            if (enableFullLanguageMode)
+            {
+                SessionState.LanguageMode = PSLanguageMode.FullLanguage;
+            }
+
+            if (setLockdownMode)
+            {
+                Environment.SetEnvironmentVariable("__PSLockdownPolicy", "0x80000007", EnvironmentVariableTarget.Machine);
+            }
+
+            if (revertLockdownMode)
+            {
+                Environment.SetEnvironmentVariable("__PSLockdownPolicy", null, EnvironmentVariableTarget.Machine);
+            }            
+        }
+    }
+'@
+
+    if (-not (Get-Command Invoke-LanguageModeTestingSupportCmdlet -ea Ignore))
+    {
+        $moduleName = Get-RandomFileName
+        $moduleDirectory = join-path $TestDrive\Modules $moduleName
+        if (-not (Test-Path $moduleDirectory))
+        {
+            mkdir $moduleDirectory -Force | Out-Null
+        }
+
+        try 
+        { 
+            Add-Type -TypeDefinition $code -OutputAssembly $moduleDirectory\TestCmdletForConstrainedLanguage.dll -ErrorAction Ignore 
+        } catch {}

```

Comment:
if this actually fails, you'll not be able to run any tests, correct? Should the catch really be silent?

Reply:
No, the failure will be at import-module

Reply:
maybe add a comment why it's ok to not do anything with the exception.